src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -22,4 +24,30 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public int hashCode() {
+        int result = 5;
+        result = 17 * result + (width == null ? 0 : width.hashCode())
+                + (length == null ? 0 : length.hashCode())
+                + (color == null ? 0 : color.hashCode());
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object rectangle) {
+        if (rectangle == this) {
+            return true;
+        }
+        if (rectangle == null) {
+            return false;
+        }
+        if (rectangle.getClass().equals(Rectangle.class)) {
+            Rectangle currentRectangle = (Rectangle) rectangle;
+            return Objects.equals(this.width, currentRectangle.width)
+                    && Objects.equals(this.length, currentRectangle.length)
+                    && Objects.equals(this.color, currentRectangle.color);
```

Comment:
https://mate-academy.github.io/jv-program-common-mistakes/java-core/equals-and-hashcode/equals-and-hashcode